src/NavigationBar/index.js

```diff
@@ -0,0 +1,214 @@
+import React, { useState } from 'react'
+import { Platform, View, Image } from 'react-native'
+import Icon from 'react-native-vector-icons/MaterialIcons'
+import { DeviceBreakpoint } from '@adalo/constants'
+
+import { Title } from './Title'
+import { MenuItems } from './MenuItems'
+import titlePlaceholder from './nav-title-image-placeholder.png'
+
+const NavigationBar = ({
+  backgroundColor,
+  shadow,
+  border,
+  borderColor,
+  borderWidth,
+  title,
+  menuItems,
+  firstMenuItem,
+  secondMenuItem,
+  thirdMenuItem,
+  fourthMenuItem,
+  fifthMenuItem,
+  profileImage,
+  _fonts,
+  editor,
+  _width,
+}) => {
+  const [activeMenuItem, setActiveMenuItem] = useState(
+    menuItems.defaultActiveMenuItem
+  )
+
+  const items = [
+    firstMenuItem,
+    secondMenuItem,
+    thirdMenuItem,
+    fourthMenuItem,
+    fifthMenuItem,
+  ]
+
+  const getBorderStyle = () => {
+    if (border) {
+      return {
+        borderBottomWidth: borderWidth,
+        borderBottomColor: borderColor,
+      }
+    }
+  }
+
+  const renderProfileImage = () => {
+    const { enabled, image, rounding, mobileText, actions } = profileImage
+    if (!enabled) {
+      return <View />
+    }
+
+    return (
+      <Image
+        source={image || titlePlaceholder}
+        style={{
+          width: 40,
+          height: 40,
+          borderRadius: rounding,
+          resizeMode: 'cover',
+          justifyContent: 'flex-end',
+        }}
+        onPress={actions}
+      />
+    )
+  }
+
+  const renderAdditionalItems = () => {
+    return <View />
+  }
+
+  const renderNavBar = () => {
+    let variant =
+      _width > DeviceBreakpoint.TABLET_BREAKPOINT ? 'desktop' : 'mobile'
+
+    let containerStyles = {
```

Comment:
There are lots of styles in here, we should look into a cleaner solution.
Also react-native has some optimizations when styles are set outside of the object and in a stylesheet
and only dynamic styles are added in the array and removed

Reply:
Also using Stylesheet.create() gives performance benefits, 
so non changing styles should be moved to one when possible

Reply:
You can also use StyleSheet.flatten or Stylesheet.compose 
to combine dynamic styles with precreated ones
I would suggest reading a bit around that before moving everything, might be worth the read